core/trino-main/src/main/java/io/trino/server/ui/UiBasicQueryInfo.java

```diff
@@ -0,0 +1,222 @@
+    private final boolean scheduled;
+    private final URI self;
+    private final String query; // trim to 300 characters
+    private final Optional<String> updateType;
+    private final Optional<String> preparedQuery;
+    private final BasicQueryStats queryStats;
+    private final ErrorType errorType;
+    private final ErrorCode errorCode;
+    private final Optional<QueryType> queryType;
+
+    @JsonCreator
+    public UiBasicQueryInfo(
+            @JsonProperty("queryId") QueryId queryId,
+            @JsonProperty("user") String user,
+            @JsonProperty("principal") Optional<String> principal,
+            @JsonProperty("source") Optional<String> source,
+            @JsonProperty("resourceGroupId") Optional<ResourceGroupId> resourceGroupId,
+            @JsonProperty("state") QueryState state,
+            @JsonProperty("memoryPool") MemoryPoolId memoryPool,
+            @JsonProperty("scheduled") boolean scheduled,
+            @JsonProperty("self") URI self,
+            @JsonProperty("query") String query,
+            @JsonProperty("updateType") Optional<String> updateType,
+            @JsonProperty("preparedQuery") Optional<String> preparedQuery,
+            @JsonProperty("queryStats") BasicQueryStats queryStats,
+            @JsonProperty("errorType") ErrorType errorType,
+            @JsonProperty("errorCode") ErrorCode errorCode,
+            @JsonProperty("queryType") Optional<QueryType> queryType)
+    {
+        this.queryId = requireNonNull(queryId, "queryId is null");
+        this.user = requireNonNull(user, "user is null");
+        this.principal = principal;
+        this.source = source;
+        this.resourceGroupId = requireNonNull(resourceGroupId, "resourceGroupId is null");
+        this.state = requireNonNull(state, "state is null");
+        this.memoryPool = memoryPool;
+        this.errorType = errorType;
+        this.errorCode = errorCode;
+        this.scheduled = scheduled;
+        this.self = requireNonNull(self, "self is null");
+        this.query = requireNonNull(query, "query is null").substring(0, Math.min(query.length(), MAX_QUERY_LENGTH - 1));
+        this.updateType = requireNonNull(updateType, "updateType is null");
+        this.preparedQuery = requireNonNull(preparedQuery, "preparedQuery is null");
+        this.queryStats = requireNonNull(queryStats, "queryStats is null");
+        this.queryType = requireNonNull(queryType, "queryType is null");
+    }
+
+    public UiBasicQueryInfo(BasicQueryInfo queryInfo)
+    {
+        this(queryInfo.getQueryId(),
+                queryInfo.getSession().getUser(),
+                queryInfo.getSession().getPrincipal(),
+                queryInfo.getSession().getSource(),
+                queryInfo.getResourceGroupId(),
+                queryInfo.getState(),
+                queryInfo.getMemoryPool(),
+                queryInfo.isScheduled(),
+                queryInfo.getSelf(),
+                queryInfo.getQuery(),
```

Comment:
Nit. Feels like query truncation fits better as a step in converting the `BasicQueryInfo`. The constructor could stay "dumb".